kernel/memory/src/lib.rs

```diff
@@ -99,6 +114,25 @@ pub fn create_contiguous_mapping<F: Into<PteFlagsArch>>(
 }
 
 
+/// A convenience function that creates a new memory mapping by allocating specific frames from physical memory.
+/// Returns the new `MappedPages`.
```

Comment:
can simplify a bit. Either way, we ought to mention that the pages it allocates are random and not specifiable when using this function.
```suggestion
/// A convenience function that maps randomly-allocated pages to the given range of frames.
```